tools/logger/convert.c

```diff
@@ -131,14 +131,53 @@ static const char *format_uid(uint32_t uid_ptr, int use_colors, bool be, bool up
 	return str;
 }
 
+/* fmt should point '%pUx`, return pointer to UUID string or zero */
+static const char *asprintf_uuid(const char *fmt, uint32_t uuid_key, bool use_colors, int *fmt_len)
+{
+	const char *fmt_end = fmt + strlen(fmt);
+	int be, upper;
+	int len = 4; /* assure full formating, with x */
+
+	if (fmt + 2 >= fmt_end || fmt[1] != 'p' || fmt[2] != 'U') {
+		*fmt_len = 0;
+		return 0;
+	}
+	/* check 'x' value */
+	switch (fmt + 3 < fmt_end ? fmt[3] : 0) {
+	case 'b':
+		be = 1;
+		upper = 0;
+		break;
+	case 'B':
+		be = 1;
+		upper = 1;
+		break;
+	case 'l':
+		be = 0;
+		upper = 0;
+		break;
+	case 'L':
+		be = 0;
+		upper = 1;
+		break;
+	default:
+		be = 0;
+		upper = 0;
+		--len;
+		break;
+	}
+	*fmt_len = len;
+	return format_uid(uuid_key, use_colors, be, upper);
```

Comment:
Does format_uid allocate a new string on its own, to obey the asprintf semantics?

Reply:
yes, format_uid use asprintf internally.